examples/compile-groovy/build.gradle

```diff
@@ -0,0 +1,59 @@
+/*
+ * This file was generated by the Gradle 'init' task.
+ *
+ * This generated file contains a sample Groovy library project to get you started.
+ * For more details on building Java & JVM projects, please refer to https://docs.gradle.org/8.3/userguide/building_java_projects.html in the Gradle documentation.
+ */
+
+plugins {
```

Comment:
I like the idea of providing a CLI entry point to this library, that makes sense to me, even if it's main purpose was to be embedded into another project.  But adding an entire Gradle distribution doesn't make sense. I find that confusing (and it might confuse some IDE as well), and I don't want to have to maintain a second distribution in an "examples" folder.

The library itself doesn't compiles the Groovy source code, it only generates it.

The idea is that the process would generate the file into a folder that is a source folder to another (Groovy) project, and then that project would see the changes  and re-compile the files into classes and create a final JAR file with those classes.  That's how I use that library.

I am fine if you want to use this library differently, in a way that is more convenient to you, but I am not willing to add and maintain a second Gradle distribution for that purpose. That is outside the scope of this library.

If you want, you can create your own GitHub project that demonstrates how to setup a project that uses this library to do a translation (including the compilation); I don't have a problem with that, and you are welcome to reference that project from the readme file you added. That's what I did a long time ago to demonstrate how to use the validation engine (https://github.com/depryf/validation-example).  But that will be your project, and you will have to maintain the Gradle distribution that goes with it. 

I don't have an issue with keeping the CLI changes, they are good changes. But I never thought a CLI was useful because the library is not very useful without the compilation step, which usually requires it to be embedded into another project.  Not to mention that most libraries are released on Maven Central nowadays and this example distribution only provides the compilation and JAR creation, not the distribution. 

Let me know what you think of all this.

Reply:
Sounds good to me! I will split off the bytecode compilation example into a separate project.